### PR TITLE
add option for config_source and deduplicate

### PR DESCRIPTION
**Pull request by aboe76**
merged file based configuration and pillar based configuration into single state.

### CODE REVIEW

File: systemd/resolved/init.sls [hunk 1]
```diff
@@ -1,35 +1,2 @@
...
-resolv.conf:
-  file.symlink:
-    - name: {{ resolved.resolv_file }}
-    - target: {{ resolved.resolv_target }}
-    - force: True
-    - backupname: /etc/resolv.conf.bak
-    - listen_in:
-      - service: resolved
+include:
+  - config
```
Comment (myii) [inline systemd/resolved/init.sls]: `config` => `.config`

File: systemd/timesyncd/init.sls [hunk 2]
```diff
@@ -1,52 +1,2 @@
...
-    - name: /etc/systemd/system/systemd-timesyncd.service.d/allowvirtual.conf
-    - contents: "[Unit]\nConditionVirtualization="
-    - makedirs: True
-{%- else %}
-  file.absent:
-    - name: /etc/systemd/system/systemd-timesyncd.service.d/allowvirtual.conf
-{%- endif %}
-    - watch_in:
-      - cmd: reload_systemd_configuration
+  - config
```
Comment (myii) [inline systemd/timesyncd/init.sls]: `config` => `.config`

File: systemd/resolved/init.sls (hunk 1 above)
Comment (aboe76) [reply systemd/resolved/init.sls]: fixed

File: systemd/timesyncd/init.sls (hunk 2 above)
Comment (aboe76) [reply systemd/timesyncd/init.sls]: fixed.

**Comment by myii** [comment]
Merged, thanks @aboe76!

**Pull request merged**